windows/test-app.js

```diff
@@ -0,0 +1,258 @@
+            if (err) {
+                throw err;
+            }
+        });
+    } else {
+        // Text file
+        const srcPermissions = fs.statSync(srcPath).mode;
+        let content = resolveContents(srcPath, replacements);
+        fs.writeFile(
+            fullDestPath,
+            content,
+            {
+                encoding: "utf8",
+                mode: srcPermissions,
+            },
+            (err) => {
+                if (err) {
+                    throw err;
+                }
+            }
+        );
+    }
+}
+
+function copyAndReplaceAll(
+    srcPath,
+    destPath,
+    relativeDestDir,
+    replacements = {}
+) {
+    return Promise.all(
+        walk(srcPath).map((absoluteSrcFilePath) => {
+            const filename = path.relative(srcPath, absoluteSrcFilePath);
+            const relativeDestPath = path.join(relativeDestDir, filename);
+            return copyAndReplace(
+                absoluteSrcFilePath,
+                destPath,
+                relativeDestPath,
+                replacements
+            );
+        })
+    );
+}
+
+const cppSourceFilesExtensions = [".h", ".cpp", ".idl", ".xaml"];
+
+function linkSourceFiles(srcPath, destPath, relativeDestDir) {
+    return Promise.all(
+        walk(srcPath).map((absoluteSrcFilePath) => {
+            if (
+                cppSourceFilesExtensions.indexOf(path.extname(absoluteSrcFilePath)) !=
+                -1
+            ) {
+                const filename = path.relative(srcPath, absoluteSrcFilePath);
+                const fullDestPath = path.join(destPath, relativeDestDir, filename);
+
+                //Avoid creating links when they already exist
+                if (!fs.existsSync(fullDestPath)) {
+                    fs.link(absoluteSrcFilePath, fullDestPath, (err) => {
+                        if (err) {
+                            throw err;
+                        }
+                    });
+                }
+            }
+        })
+    );
+}
+
+async function copyProjectTemplateAndReplace(
+    srcRootPath,
+    destPath,
+    reactNativeModulePath,
+    testAppNodeModulePath
+) {
+    if (!srcRootPath) {
+        throw new Error("Need a path to copy from");
+    }
+
+    if (!destPath) {
+        throw new Error("Need a path to copy to");
+    }
+
+    if (!reactNativeModulePath) {
+        throw new Error("react-native-windows node module is not installed");
+    }
+
+    if (!testAppNodeModulePath) {
+        throw new Error("react-native-test-app node module is not installed");
+    }
+
+    const projDir = "ReactTestApp";
+
+    fs.mkdirSync(path.join(destPath, windowsDir, projDir, "Bundle"), {
+        recursive: true,
+    });
+
+    const manifestFilePath = findClosestPathTo("app.json");
+
+    //Read path to resources from manifest and copy them
+    fs.readFile(manifestFilePath, function (error, content) {
+        var resourcesPaths = JSON.parse(content.toString()).resources;
```

Comment:
Prefer `let` over `var`:
```suggestion
        let resourcesPaths = JSON.parse(content.toString()).resources;
```